### PR TITLE
Also test with Java 17 and version updates

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

**Comment by Ao-senXiong** [comment]
@wmdietl Could you invite me to this org? I can start look into this PR.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -12,22 +12,23 @@ jobs:
   build:
     strategy:
       matrix:
-        jdk: [ 8, 11 ]
+        jdk: [ 8, 11, 17, 21 ]
```
Comment (wmdietl) [inline .github/workflows/main.yml:15]:
```suggestion
        jdk: [ 8, 11, 17, 21, 22 ]
```

**Comment by wmdietl** [comment]
@Ao-senXiong Can't you open PRs at the moment?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -12,22 +12,23 @@ jobs:
   build:
     strategy:
       matrix:
-        jdk: [ 8, 11 ]
+        jdk: [ 8, 11, 17, 21, 22 ]
```
Comment (wmdietl) [inline .github/workflows/main.yml:15]:
```suggestion
        jdk: [ 8, 11, 17 ]
```

Comment (wmdietl) [reply .github/workflows/main.yml:15]: @Ao-senXiong #15 is basically this. I'll merge this PR and in your PR please look into support for JDK 21.

**Pull request merged**